web/virtual_lab/static/virtual_lab/js/code_editor.js

```
@@ -0,0 +1,69 @@
+    body: JSON.stringify(payload)
+  })
+  .then(res => {
+    console.log("⬅️ HTTP status:", res.status);
+    return res.text().then(text => {
+      console.log("⬅️ Raw response text:", text);
+      try {
+        return JSON.parse(text);
+      } catch (err) {
+        throw new Error("Invalid JSON: " + err.message);
+      }
+    });
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)